TestLogBrokerNoFollow fails sometimes

```
100 received
200 received
300 received
400 received
500 received
600 received
700 received
800 received
900 received
1000 received
1100 received
1200 received
1300 received
1400 received
1500 received
1600 received
1700 received
1800 received
1900 received
2000 received
2100 received
2200 received
2300 received
2400 received
2500 received
2600 received
2700 received
2800 received
2900 received
3000 received
3100 received
3200 received
3300 received
3400 received
3500 received
3600 received
3700 received
3800 received
3900 received
4000 received
4100 received
4200 received
4300 received
4400 received
4500 received
--- FAIL: TestLogBrokerNoFollow (5.11s)
	Error Trace:	broker_test.go:170
			broker_test.go:427
	Error:		subscription expected
```

ping @aluzzardi

--- a/manager/logbroker/broker.go
+++ b/manager/logbroker/broker.go
@@ -173,7 +173,7 @@ func (lb *LogBroker) watchSubscriptions(nodeID string) ([]*subscription, chan ev
 	}))
 
 	// Grab current subscriptions.
-	subscriptions := make([]*subscription, 0, len(lb.registeredSubscriptions))
+	subscriptions := make([]*subscription, 0)
 	for _, s := range lb.registeredSubscriptions {
 		if s.Contains(nodeID) {
 			subscriptions = append(subscriptions, s)

--- a/manager/logbroker/broker_test.go
+++ b/manager/logbroker/broker_test.go
@@ -138,11 +138,9 @@ func listenSubscriptions(ctx context.Context, t *testing.T, client api.LogBroker
 	require.NoError(t, err)
 
 	ch := make(chan *api.SubscriptionMessage)
-	ready := make(chan struct{})
 	go func() {
 		defer close(ch)
 
-		close(ready)
 		for {
 			select {
 			case <-ctx.Done():
@@ -157,7 +155,6 @@ func listenSubscriptions(ctx context.Context, t *testing.T, client api.LogBroker
 		}
 	}()
 
-	<-ready
 	return ch
 }
 
@@ -409,6 +406,16 @@ func TestLogBrokerNoFollow(t *testing.T) {
 
 		return nil
 	}))
+
+	// We need to sleep here to give ListenSubscriptions time to call
+	// registerSubscription before SubscribeLogs concludes that one or both
+	// of the agents are not connected, and prematurely calls Done for one
+	// or both nodes. Think of these stream RPC calls as goroutines which
+	// don't have synchronization around anything that happens in the RPC
+	// handler before a send or receive. It would be nice if we had a way
+	// of confirming that a node was listening for subscriptions before
+	// calling SubscribeLogs, but the current API doesn't provide this.
+	time.Sleep(time.Second)
 
 	// Subscribe to logs in no follow mode
 	logs, err := client.SubscribeLogs(ctx, &api.SubscribeLogsRequest{

